### PR TITLE
Issue 4996

### LINKED ISSUE #4996: Avails Function
- [ ] Create a form specific for the avails (it should be separated from the search form).
- [ ] Implement this function in the component on titles coming out of Algolia (we filter the avail locally after applying other filters from Algolia).
- [ ] Remove search limitation from algolia to query all the titles that match the query (else avails will apply only on the first titles) and remove the load more button

### PR DESCRIPTION
**Pull request by fritzschoff**
close #4996

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts [hunk 1]
```diff
@@ -63,30 +72,40 @@ export class ListComponent implements OnInit, OnDestroy {
       }
     }
 
-    this.sub = this.searchForm.valueChanges.pipe(startWith(this.searchForm.value),
+    const mandates = await this.contractService.getValue(ref => ref.where('type', '==', 'mandate'));
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts:75]: You don't need this anymore, you're already doing it in the getTerms method

File: apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts [hunk 2]
```diff
@@ -63,30 +70,42 @@ export class ListComponent implements OnInit, OnDestroy {
...
+    const sales = await this.contractService.getValue(ref => ref.where('type', '==', 'sale'));
+
+    this.mandateTerms = await (await Promise.all(mandates.map(mandate =>
+      this.termService.getValue(mandate.termIds)))).flat() as Term<Date>[];
+    this.salesTerms = await (await Promise.all(sales.map(sale =>
+      this.termService.getValue(sale.termIds)))).flat() as Term<Date>[];
+
+    this.sub = combineLatest([
+      this.searchForm.valueChanges.pipe(startWith(this.searchForm.value)),
```
[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts:82]: You don't want to do that ?

**Review by GrandSchtroumpf — APPROVED** [review]

File: apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fritzschoff) [reply apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts:82]: I don'T really see any gain?

File: apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts [hunk 3]
```diff
@@ -40,20 +44,25 @@ export class ListComponent implements OnInit, OnDestroy {
   public hitsViewed = 0;
 
   private sub: Subscription;
-  private loadMoreToggle: boolean;
-  private lastPage: boolean;
+
+  private terms: {
+    mandateTerms: Term<Date>[]
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts:49]: I would rename :
```typescript
terms {
  mandate: Term<Date>[],
  sales: Term<Date>[]
}
```

**Pull request merged**